iothub_client/src/iothub_client_core_ll.c

```diff
@@ -833,6 +833,13 @@ static IOTHUB_CLIENT_CORE_LL_HANDLE_DATA* initialize_iothub_client(const IOTHUB_
                     config = client_config;
                 }
             }
+            else if (device_config == NULL)
```

Comment:
Is this right?? on lines 775+ it looks like device_config == NULL is a valid scenario.

Reply:
without it, it would fall into the else and at line 847 cause an AV

Reply:
fixed